### PR TITLE
Batch origins/destinations

### PR DESCRIPTION
**Pull request by Wizard-Rogue**
I modified the GET function to accept the parameters origins and destinations to accommodate batch fetching of data.

### CODE REVIEW

**Comment by edwlook** [comment]
@Wizard-Rogue Thanks for the pull request! I added some suggestions below that will help us move forward.

File: index.js [hunk 1]
```diff
@@ -51,30 +63,41 @@ GoogleDistance.prototype.formatOptions = function(args) {
...
+  var results = [];
+
+  for (var i = data.rows.length - 1; i >= 0; i--) {
+    var element = data.rows[i].elements[0];
+    var resultStatus = element.status;
+    if (resultStatus != 'OK') {
+      return callback(new Error('Result error: ' + resultStatus));
+    };
+    element.origin = data.origin_addresses[i];
+    element.destination = data.destination_addresses[i];
```
Comment (edwlook) [inline index.js]: For batch requests, the Google distance API actually returns distance data for each origin/destination combination. This is behavior we should replicate in our code.

See here for more details: https://developers.google.com/maps/documentation/distancematrix/#DistanceMatrixResponses

File: test/index.test.js [hunk 2]
```diff
@@ -54,6 +62,56 @@ describe('GoogleDistance', function() {
...
+          origin: 'San Diego, CA, USA',
+          destination: 'Seattle, WA, USA',
+          mode: 'driving',
+          units: 'metric',
+          language: 'en',
+          avoid: null,
+          sensor: false
+        }];
+
+        for (var i in expectedData) {
```
Comment (edwlook) [inline test/index.test.js]: It's best not use `for..in..` to iterate over an array.
Here's why: http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea

Instead use a regular `for` loop: `for (int i = 0; i < expectedData.length; i++) {...}`

File: index.js [hunk 3]
```diff
@@ -24,6 +24,15 @@ GoogleDistance.prototype.get = function(args, callback) {
...
 
 GoogleDistance.prototype.formatOptions = function(args) {
+  var pipeIt = function (array) {
+    var string = '';
+    for (var i = array.length - 1; i >= 0; i--) {
+      string += array[i] + '|';
+    };
+    string = string.substring(0, string.length - 1);
+    return string;
+  };
```
Comment (edwlook) [inline index.js]: `array.join('|')` does the same thing as this function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join

**Comment by Wizard-Rogue** [comment]
I see. Thanks for the notes. I'll apply them as soon as I can.

**Comment by edwlook** [comment]
@Wizard-Rogue Thanks for the changes!

File: index.js [hunk 4]
```diff
@@ -51,30 +54,44 @@ GoogleDistance.prototype.formatOptions = function(args) {
...
     return callback(new Error('Status error: ' + requestStatus + ': ' + data.error_message));
   }
-  var element = data.rows[0].elements[0];
-  var resultStatus = element.status;
-  if (resultStatus != 'OK') {
-    return callback(new Error('Result error: ' + resultStatus));
-  }
+  var results = [];
+
+  for (var i = 0; i <= data.origin_addresses.length - 1; i++) {
```
Comment (edwlook) [inline index.js]: Here you can use: `for (var i = 0; i < data.origin_addresses.length; i++) {...}`
It's just a bit easier to read.

File: index.js [hunk 5]
```diff
@@ -51,30 +54,44 @@ GoogleDistance.prototype.formatOptions = function(args) {
...
+    for (var j = 0; j <= data.destination_addresses.length - 1; j++) {
```
Comment (edwlook) [inline index.js]: Same as above.

**Comment by Wizard-Rogue** [comment]
Got it. Changes have been made.

**Comment by Wizard-Rogue** [comment]
@edwlook, I noticed that Google attempts to convert any coordinates into readable addresses. Would you mind it if I added the fields original_origin and original_destination to the results? It would be helpful to the people who are using coordinates as their input.

**Comment by edwlook** [comment]
@Wizard-Rogue Good idea. Can you make a new pull request for that after this one is merged?

**Pull request merged**